### ISSUE TITLE
Missing context fields on first fetch

### ISSUE THREAD

**Issue by marhaupe** [Unleash/unleash_proxy_client_flutter#18]
### Describe the bug

Roughly, I'm doing the following things to initialize my unleash client instance:

```dart

class UnleashService {
  static Future<UnleashClient> create() async {
    PackageInfo packageInfo = await PackageInfo.fromPlatform();

    int buildNumber = int.parse(packageInfo.buildNumber);
    String version = packageInfo.version;

    final client = UnleashClient(
        url: Uri.parse(API_URL), clientKey: API_TOKEN, appName: APP_NAME);

    Completer<UnleashClient> completer = Completer<UnleashClient>();

    client.on('ready', (_) {
      if (completer.isCompleted) return;
      kL.i("unleash client ready");
      completer.complete(client);
    });

    client.on('error', (data) {
      if (completer.isCompleted) return;
      kL.e('error initializing unleash client: $data');
      completer.complete(client);
    });

    client.setContextField('buildNumber', buildNumber.toString());
    client.setContextField('version', version);
    client.start();

    return completer.future;
  }
}


```

So I'm setting context fields, starting, and then the client should be in the ready state soon after. Looking at the code, when running `client.setContextField`, the client awaits until the `readyEvent` has been emitted once before the context fields are set. For us, this 'wastes' the first feature toggle query because our strategies heavily rely on the fields in the context. Is there any particular reason as to why the client needs to wait before setting the context? I tried just mutating `client.context.properties` directly before running `client.start` and that seems to have done the trick, but I'm not sure if I'm missing something.

### Subscription type

None

### Hosting type

None

**Comment by kwasniew** [comment]
Can you try initializing your client like this test:   [test('update context without await', () async {](https://github.com/Unleash/unleash_proxy_client_flutter/blob/main/test/unleash_proxy_client_flutter_test.dart#L580) and let me know if it behaves as you'd expect?

When I have some spare time I'd try to replicate your use case in a unit test.

**Comment by marhaupe** [comment]
I actually found that in the meantime and it seems like that works. The first request now includes the custom context properties. 

However, I've since stumbled onto the next problem, maybe you can help me out here as well. The URL looks like this: `/api/frontend?sessionId=283815071&buildNumber=1&version=4.2.3`. In comparison, the js proxy client sends requests to this url: `/api/frontend?sessionId=807661975&properties%5Bversion%5D=4.2.3`. I'm trying to set up Unleash Edge, and only the latter request works. I see that there are test cases for the former format (https://github.com/Unleash/unleash_proxy_client_flutter/blob/f281671aa1543726c2a943c96282ecc47163bd64/test/unleash_proxy_client_flutter_test.dart#L496C17-L496C17), so I don't know if it's an Unleash Edge specific problem or something else.

**Comment by kwasniew** [comment]
I think what the flutter SDK is doing is fine. I also tested locally with unleash-proxy-js-client talking directly to the unleash-server at api/frontend and the query params were flat.
So:
`unleash.updateContext({userId: '1'});`
translated to `http://localhost:4242/api/frontend?environment=default&appName=my-local-webapp&userId=1`

**Comment by kwasniew** [comment]
Full code I used for testing:
```js
const unleash = new UnleashClient({
    url: 'http://localhost:4242/api/frontend',
    clientKey: 'default:development.ab1c6.....',
    appName: 'my-local-webapp',
    environment: 'default',
    metricsInterval: 10,
    refreshInterval: 1,
});

unleash.updateContext({userId: '1'});

unleash.start();


unleash.on('ready', async () => {
    // eslint-disable-next-line no-constant-condition
    while(true) {
        console.log(unleash.isEnabled('teststrategies'));
        await new Promise((resolve) => {
            setTimeout(() => {resolve('done')}, 1000);
        });
    }
});
```

**Comment by kwasniew** [comment]
It looks to me like you did this in the JS world (extra properties field):
```
unleash.updateContext({properties: {userId: '1'}});
```

**Comment by marhaupe** [comment]
Exactly, I need to use the extra properties field. The JS snippet behaves the same way, so this works:
```js
unleash.updateContext({
  properties: {
    version: "5.0.0",
  },
});
```

While this does not work:
```js
unleash.updateContext({
  version: "5.0.0",
});
```

If I skip the Unleash Edge instance and set my `url` directly to my Unleash API, both variants work.

**Comment by kwasniew** [comment]
Ok so it looks like the bug is in the flutter SDK. I will try to patch it. Basically all non-standard user properties should be encoded in the query params with the properties prefix.

**Comment by marhaupe** [comment]
Wow that was quick, I'm still building my app with your branch of the lib. I'll let you know! Thanks a lot, really appreciate the quick response.

**Comment by marhaupe** [comment]
Awesome, works like a charm. I can use both Unleash Edge and the hosted API interchangeably 🙌

**Comment by kwasniew** [comment]
Glad to hear it. And thanks for helping with making this library more correct :)
